src/main/java/reactor/netty/udp/UdpClientConfig.java

```diff
@@ -74,15 +74,26 @@ public final InternetProtocolFamily family() {
 	}
 
 	@Override
-	protected ChannelFactory<? extends Channel> connectionFactory(EventLoopGroup elg) {
-		ChannelFactory<DatagramChannel> channelFactory;
+	protected Class<? extends Channel> channelType(boolean isDomainSocket) {
+		if (isDomainSocket) {
+			throw new UnsupportedOperationException();
+		}
+		return DatagramChannel.class;
+	}
+
+	@Override
+	protected ChannelFactory<? extends Channel> connectionFactory(EventLoopGroup elg, boolean isDomainSocket) {
+		if (isDomainSocket) {
+			throw new UnsupportedOperationException();
+		}
+		DatagramChannel channel;
 		if (isPreferNative()) {
-			channelFactory = () -> loopResources().onChannel(DatagramChannel.class, elg);
+			channel = loopResources().onChannel(DatagramChannel.class, elg);
```

Comment:
This changes the semantics of when objects are created (once vs at factory invocation). Is this intended?

Reply:
thanks for spotting this -> changed